woocommerce/woocommerce-ios
# [Custom amounts M4] Taxes percentage new design

Sorry for the large PR. Much of it is refactors, e.g. moving classes to other files, but I'm happy to split this PR in smaller ones if required.

Closes: #11220 


## Description

With this PR we implement the new design for Percentages in custom amounts. Here we:

- Refactor the AddCustomAmountsViewModel and view to implement the new design.
- Implement options dialog with the bottom sheet in `OrderCustomAmountsSection`
- Refactor the `EditableOrderViewModel` for the new case
- Add unit tests

## Testing instructions


1. Go to order
2. Tap on + to create a new order
3. Add a custom amount. See that you see no options dialog, and you go directly to add the custom amount with fixed amount.
4. When you're back to the order creation screen, tap to add a new amount. See that you see the dialog. Tap on percentage.
5. See that you can add a custom amount based on the order total percentage.
6. When you are back to the order details screen, tap to edit a custom amount. See that in both cases, fixed amount and percentage, data is filled.


## Screenshots


https://github.com/woocommerce/woocommerce-ios/assets/1864060/c7187ac1-9186-46f6-97f3-02a9e5c75656

---
